### PR TITLE
Use a simpler code path for np.pad with mode='wrap'

### PR DESCRIPTION
**Pull request by shoyer**
This code path avoids any calls to lax.rev(), and seems to make a small but
measurable performance improvement for some of use cases.

### CODE REVIEW

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1465,18 +1472,31 @@ def _pad(array, pad_width, mode, constant_values):
       widths[i] = (0, pad_width[i, 1], 0)
       array = lax.pad(array, constant_values[i, 1], widths)
     return array
-  elif mode in ("symmetric", "reflect", "wrap"):
+  elif mode == "wrap":
+    for i in xrange(nd):
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py]: Shouldn't this be `range`, or won't you need `six.moves` import of `xrange`?

**Review by hawkinsp — APPROVED** [review]
I'm a bit skeptical that there are any performance differences here, given the implementation of `_pad` is under a `jit`. If there are, that points to a missing XLA optimization and we should probably file a bug for it.

**Comment by shoyer** [comment]
On TPU, I'm seeing a difference between 3.43 seconds vs 2.76 seconds per step for my model. The slow down on CPU is also present but slower.

Here's a synthetic example that reproduces the issue:
```python
import jax.numpy as np
import jax
from jax import lax

def f_old_pad(x):
  # something laplacian like
  padded = np.pad(x, [(1, 1), (0, 0)], mode='wrap')
  return 2 * x - padded[2:] - padded[:-2]

def f_new_pad(x):
  # something laplacian like
  padded = np.concatenate([x[-1:], x, x[:1]])
  return 2 * x - padded[2:] - padded[:-2]

def repeated(f, num_times):
  @jax.jit
  def g(x):
    return lax.scan(lambda c, _: (f(c), None), x, np.arange(num_times))
  return g

x = jax.device_put(np.zeros((256, 256)))
loop_old = repeated(f_old_pad, 2000)
loop_new = repeated(f_new_pad, 2000)

jax.device_get(loop_old(x))
%time jax.device_get(loop_old(x))
# Wall time: 107 ms

jax.device_get(loop_new(x))
%time jax.device_get(loop_new(x))
# Wall time: 79.2 ms
```

File: jax/numpy/lax_numpy.py (hunk 1 above)
Comment (shoyer) [reply jax/numpy/lax_numpy.py]: I just adapted this from what you already had. It looks like this `xrange` is already `six.moves.xrange`.

**Pull request merged**